assets/js/googlesitekit/datastore/modules/modules.js

```diff
@@ -0,0 +1,434 @@
+	 * ```
+	 * slug: {
+	 *   "slug": "tagmanager",
+	 *   "name": "Tag Manager",
+	 *   "description": "Tag Manager creates an easy to manage way to create tags on your site without updating code.",
+	 *   "homepage": "https://tagmanager.google.com/",
+	 *   "internal": false,
+	 *   "active": false,
+	 *   "connected": false,
+	 *   "dependencies": [
+	 *     "analytics"
+	 *   ],
+	 *   "dependants": []
+	 * }
+	 * ```
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param {Object} state Data store's state.
+	 * @return {Object|undefined} Modules available on the site.
+	 */
+	getModules( state ) {
+		const { modules } = state;
+
+		return modules;
+	},
+
+	/**
+	 * Get a specific module by slug.
+	 *
+	 * Returns a specific module by its slug.
+	 * Returns `undefined` if state is still loading or if said module doesn't exist.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param {Object} state Data store's state.
+	 * @param {string} slug  Module slug.
+	 * @return {Object|undefined} A specific module object.
+	 */
+	getModule: createRegistrySelector( ( select ) => ( state, slug ) => {
+		const modules = select( STORE_NAME ).getModules();
+
+		// Return `undefined` if modules haven't been loaded yet.
+		if ( modules === undefined ) {
+			return undefined;
+		}
+
+		// A module with this slug couldn't be found; return `null` to signify the
+		// "not found" state.
+		if ( modules[ slug ] === undefined ) {
+			return null;
+		}
+
+		// This module exists, so let's return it.
+		return modules[ slug ];
+	} ),
+
+	/**
+	 * Check a module's activation status.
+	 *
+	 * Returns `true` if the module exists and is active.
+	 * Returns `false` if the module exists but is not active.
+	 * Returns `undefined` if state is still loading or if no module with that slug exists.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param {Object} state Data store's state.
+	 * @param {string} slug  Module slug.
+	 * @return {Object|undefined} A specific module object.
+	 */
+	isModuleActive: createRegistrySelector( ( select ) => ( state, slug ) => {
+		const module = select( STORE_NAME ).getModule( slug );
+
+		// Return `undefined` if modules haven't been loaded yet.
+		if ( module === undefined ) {
+			return undefined;
+		}
+
+		// A module with this slug couldn't be found; return `null` to signify the
+		// "not found" state.
+		if ( module === null ) {
+			return null;
+		}
+
+		return module.active;
+	} ),
+
+	/**
+	 * Check if a module's status is changing.
+	 *
+	 * Returns `true` if the module exists and is changing its `active` flag.
+	 * Returns `false` if the module exists but is not changing its `active` flag.
+	 * Returns `undefined` if state is still loading or if no module with that slug exists.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param {Object} state Data store's state.
+	 * @param {string} slug  Module slug.
+	 * @return {boolean|undefined} Activation change status.
+	 */
+	isChangingModuleActivation: ( state, slug ) => {
```

Comment:
Maybe `isSettingModuleStatus`? This would be more in line with the other names used throughout the datastore. Both names are not that great I think, but since this is also a very specific selector, I don't have a strong opinion.